app/process/barriers/barriers.lua

```diff
@@ -0,0 +1,157 @@
+require("Set")
+require("FilterTags")
+require("Metadata")
+require("HighwayClasses")
+
+local lineBarriers = osm2pgsql.define_table({
+  name = 'lineBarriers',
+  ids = { type = 'any', id_column = 'osm_id', type_column = 'osm_type' },
+  columns = {
+    { column = 'tags', type = 'jsonb' },
+    { column = 'meta', type = 'jsonb' },
+    { column = 'geom', type = 'linestring' },
+  }
+})
+
+local excludedLineBarriers = osm2pgsql.define_table({
+  name = 'excludedLineBarriers',
+  ids = { type = 'any', id_column = 'osm_id', type_column = 'osm_type' },
+  columns = {
+    { column = 'tags', type = 'jsonb' },
+    { column = 'meta', type = 'jsonb' },
+    { column = 'geom', type = 'linestring' },
+  }
+})
+
+local areaBarriers = osm2pgsql.define_table({
+  name = 'areaBarriers',
+  ids = { type = 'any', id_column = 'osm_id', type_column = 'osm_type' },
+  columns = {
+    { column = 'tags', type = 'jsonb' },
+    { column = 'meta', type = 'jsonb' },
+    { column = 'geom', type = 'multipolygon' },
+  }
+})
+
+local excludedAreaBarriers = osm2pgsql.define_table({
+  name = 'excludedAreaBarriers',
+  ids = { type = 'any', id_column = 'osm_id', type_column = 'osm_type' },
+  columns = {
+    { column = 'tags', type = 'jsonb' },
+    { column = 'meta', type = 'jsonb' },
+    { column = 'geom', type = 'multipolygon' },
+  }
+})
+
+
+local allowedTags = Set({
+  'tunnel',
+  'waterway',
+  'aerodrome',
+  'name',
+  'natural',
+  'railway',
+  'usage',
+  'circumference',
+  'area',
+})
+
+local function isAreaBarrier(object)
+  local tags = object.tags
+  local isBarrier = false
+
+  if tags.natural == 'water' then
+    local area = object:as_multipolygon():transform(3857):area()
+    tags.area = area
+    -- TODO: IMO it would be better to have this check on the single polygons of a multipolygon
+    if area > 100000 then
+      isBarrier = true
+      tags.area = area
+    elseif object.type == 'way' then
+      local circumference = object:as_linestring():transform(3857):length()
+      tags.circumference = circumference
+      if circumference > 1000 then
+        isBarrier = isBarrier or (area / circumference) < 3
+        isBarrier = true
+      end
+    end
+  end
+
+  isBarrier = isBarrier or tags.aeroway == 'aerodrome'
+
+  return isBarrier
+end
+
+function osm2pgsql.process_way(object)
+  if object.is_closed then -- process as polygon
+    if isAreaBarrier(object) then
+      FilterTags(object.tags, allowedTags)
+      areaBarriers:insert({
+        tags = object.tags,
+        meta=Metadata(object),
+        geom=object:as_multipolygon()
+      })
+      return
+    end
+    excludedAreaBarriers:insert({
+      tags=object.tags,
+      meta=Metadata(object),
+      geom=object:as_multipolygon()
+    })
+    return
+  else --process as linestring
+    local tags = object.tags
+    if tags.tunnel =='yes' then return end -- we don't consider tunnels as barriers
+
+    local isBarrier = HighwayClasses[tags.highway]
```

Comment:
Put into excludedLineBarriers to be consistent.

```suggestion
    -- we don't consider tunnels as barriers
    local isBarrier = tags.tunnel != 'yes' and HighwayClasses[tags.highway]
```

Reply:
Good point! I thought about the exclude table of this processing and think for now it's a bit useless because we just insert every way and relation without filtering anything. So maybe we have to think of a minimal filter which we take as the base for this dataset and insert only ways and relations in to the exclude table which surpassed this basic condition.